docs/lara-plugin-api.md

```diff
@@ -0,0 +1,129 @@
+
+# LARA Plugin API
+
+A LARA Plugin architecture which can be used by
+javascript developers to add new features to LARA.
+
+NP 2018-08-29 -- This documentation is being developed and is
+subject to change.  
+
+## The hosted LARA Plugin API
+
+This describes the functions attached to the global `window.LARA` object
+LARA exports.
+
+```typescript
+
+/****************************************************************************
+// @function addSidebar: Ask lara to add a new sidebar window
+// @arg {ISidebarOptions} sidebarOptions
+// @returns A parent DomNode where component can render content
+****************************************************************************/
+LARA.addSidebar(sidebarOptions: ISidebarOptions) : DomNode
```

Comment:
If my suggestion below about providing a react class and properties works, then it should be used here too.

Also I had considered it might be better for this to be a plugin configuration instead of a API method. So when an admin adds the plugin to lara they specify:
- the url for the javascript
- optional text for a sidebar
- optional image for a sidebar
Then LARA configures the sidebar on the server side, and provides the dom element in the initialization configuration to the plugin. This is less flexible though, the plugin might not want to show the sidebar on every page. So lets try the API approach first and see how well it works.

Reply:
I think that the additional advantage of API approach is that plugins will be 100% self-contained.  Fewer things to configure on staging, production, locally, etc. E.g. if the developer is testing some plugins locally and needs to set them up first, he doesn't need to know that this one requires some sidebar config. The plugin will do it itself providing right text and icon.